### PR TITLE
Update UI built-in evaluators to version 2

### PR DESCRIPTION
**Pull request by diondrapeck**

### CODE REVIEW

File: assets/promptflow/evaluators/models/coherence-evaluator/spec.yaml [hunk 1]
```diff
@@ -5,8 +5,8 @@ properties:
   is-promptflow: true
   is-evaluator: true
   show-artifact: true
-  _default-display-file: ./evaluator/flow/prompt.jinja2
```
Comment (ninghu) [inline assets/promptflow/evaluators/models/coherence-evaluator/spec.yaml:8]: I thought for the prompt based ones, we want to show its prompt to customers by default.

File: assets/promptflow/evaluators/models/coherence-evaluator/spec.yaml [hunk 2]
```diff
@@ -5,8 +5,8 @@ properties:
...
+  _default-display-file: ./evaluator/_coherence.py
 tags:
   Preview: ""
   hiddenlayerscanned: ""
-version: 1
\ No newline at end of file
+version: 2
```
Comment (ninghu) [inline assets/promptflow/evaluators/models/coherence-evaluator/spec.yaml:12]: I think we should check all the descriptions.md, and do the qa->qr rename as needed.

File: assets/promptflow/evaluators/models/fluency-evaluator/spec.yaml [hunk 3]
```diff
@@ -5,8 +5,8 @@ properties:
   is-promptflow: true
   is-evaluator: true
   show-artifact: true
-  _default-display-file: ./evaluator/flow/prompt.jinja2
+  _default-display-file: ./evaluator/_fluency.py
```
Comment (ninghu) [inline assets/promptflow/evaluators/models/fluency-evaluator/spec.yaml:8]: same here

File: assets/promptflow/evaluators/models/meteor-score-evaluator/description.md [hunk 4]
```diff
@@ -4,4 +4,4 @@
 | What is this metric? | Evaluates the quality of the generated text by considering precision, recall, and a range of linguistic features like synonyms, stemming, and word order. |
 | How does it work? | The METEOR score is calculated based on the harmonic mean of unigram precision and recall, with higher weight given to recall. It also incorporates additional features such as stemming (matching word roots), synonym matching, and a penalty for incorrect word order. The final score ranges from 0 to 1, where 1 indicates a perfect match. |
 | When to use it? | Use the METEOR score when you want a more linguistically informed evaluation metric that captures not only n-gram overlap but also accounts for synonyms, stemming, and word order. This is particularly useful for evaluating tasks like machine translation, text summarization, and text generation. |
-| What does it need as input? | Ground Truth Response, Generated Response |
\ No newline at end of file
+| What does it need as input? | Rouge type, Ground Truth Response, Generated Response |
```
Comment (ninghu) [inline assets/promptflow/evaluators/models/meteor-score-evaluator/description.md:7]: Thanks for making this change which I missed ealier, but this change needs to go to rouge-score-evaluator.

File: assets/promptflow/evaluators/models/coherence-evaluator/spec.yaml (hunk 2 above)
Comment (diondrapeck) [reply assets/promptflow/evaluators/models/coherence-evaluator/spec.yaml:12]: Yes - all the descriptions need to be reviewed in another commit. I need Ankit's help on this since it was his work item, but he was busy Friday so I'll check with him today.

File: assets/promptflow/evaluators/models/meteor-score-evaluator/description.md (hunk 4 above)
Comment (diondrapeck) [reply assets/promptflow/evaluators/models/meteor-score-evaluator/description.md:7]: Ah yes - thanks for catching that

File: assets/promptflow/evaluators/models/coherence-evaluator/spec.yaml (hunk 2 above)
Comment (diondrapeck) [reply assets/promptflow/evaluators/models/coherence-evaluator/spec.yaml:12]: Minsoo signed off and the qa -> qr changes have been made.

File: assets/promptflow/evaluators/models/groundedness-evaluator/spec.yaml [hunk 5]
```diff
@@ -9,4 +9,4 @@ properties:
 tags:
```
Comment (ninghu) [inline assets/promptflow/evaluators/models/groundedness-evaluator/spec.yaml:9]: This path doesn't seem to be valid anymore. There should be no "flow" dir.

File: assets/promptflow/evaluators/models/hate-unfairness-evaluator/spec.yaml [hunk 6]
```diff
@@ -4,7 +4,9 @@ path: ./
 properties:
   is-promptflow: true
   is-evaluator: true
+  show-artifact: true
```
Comment (ninghu) [inline assets/promptflow/evaluators/models/hate-unfairness-evaluator/spec.yaml:7]: for content safety ones, it shouldn't show the aritfact. please revert the change

File: assets/promptflow/evaluators/models/protected-material-evaluator/spec.yaml [hunk 7]
```diff
@@ -4,6 +4,8 @@ path: ./
 properties:
   is-promptflow: true
   is-evaluator: true
+  show-artifact: true
```
Comment (ninghu) [inline assets/promptflow/evaluators/models/protected-material-evaluator/spec.yaml:7]: same here

File: assets/promptflow/evaluators/models/similarity-evaluator/spec.yaml [hunk 8]
```diff
@@ -9,4 +9,4 @@ properties:
 tags:
```
Comment (ninghu) [inline assets/promptflow/evaluators/models/similarity-evaluator/spec.yaml:9]: same here. _default-display-file path need to be updated.

File: assets/promptflow/evaluators/models/coherence-evaluator/spec.yaml (hunk 1 above)
Comment (ninghu) [reply assets/promptflow/evaluators/models/coherence-evaluator/spec.yaml]: I believe the decision to show articles or not, and what to display by default, was defined by Wenxin.

Let's keep the behavior the same as before. In general:

Prompt-based evaluators: show the prompt
Content safety evaluators: do not show the artifact
Math-based evaluators: show the code

**Comment by ninghu** [comment]
Do we miss the Retrieval evaluator in this PR?

**Comment by diondrapeck** [comment]
> Do we miss the Retrieval evaluator in this PR?

No, it's intentionally excluded. It only supports conversation, not single Q/R, so I confirmed with Minsoo it should not be a built-in evaluator.

File: assets/promptflow/evaluators/models/hate-unfairness-evaluator/spec.yaml (hunk 6 above)
Comment (diondrapeck) [reply assets/promptflow/evaluators/models/hate-unfairness-evaluator/spec.yaml:7]: So the rules are:

- no artifact for content safety evaluators
- prompt template for prompt-based evaluators
- show the implementation file (e.g. _bleu.py) for all others

Is that correct?

File: assets/promptflow/evaluators/models/groundedness-evaluator/spec.yaml (hunk 5 above)
Comment (diondrapeck) [reply assets/promptflow/evaluators/models/groundedness-evaluator/spec.yaml:9]: Good catch - I corrected the directory path.

File: assets/promptflow/evaluators/models/hate-unfairness-evaluator/spec.yaml (hunk 6 above)
Comment (ninghu) [reply assets/promptflow/evaluators/models/hate-unfairness-evaluator/spec.yaml]: Yes!

**Review by ninghu — APPROVED** [review]

**Review by vizhur — APPROVED** [review]
